disnake/client.py

```diff
@@ -760,6 +769,159 @@ def dispatch(self, event: str, *args: Any, **kwargs: Any) -> None:
         else:
             self._schedule_event(coro, method, *args, **kwargs)
 
+        for event_ in self.extra_events.get(method, []):
+            self._schedule_event(event_, method, *args, **kwargs)
+
+    def add_listener(self, func: CoroFunc, name: Union[str, Event] = MISSING) -> None:
+        """The non decorator alternative to :meth:`.listen`.
+
+        .. versionchanged:: 2.10
+            The definition of this method was moved from the :class:`.ext.commands.Bot`
+            base class to :class:`.Client` class.
+
+        Parameters
+        ----------
+        func: :ref:`coroutine <coroutine>`
+            The function to call.
+        name: Union[:class:`str`, :class:`.Event`]
+            The name of the event to listen for. Defaults to ``func.__name__``.
+
+        Example
+        --------
+
+        .. code-block:: python
+
+            async def on_ready(): pass
+            async def my_message(message): pass
+            async def another_message(message): pass
+
+            client.add_listener(on_ready)
+            client.add_listener(my_message, 'on_message')
+            client.add_listener(another_message, Event.message)
+
+        Raises
+        ------
+        TypeError
+            The function is not a coroutine or a string or an :class:`.Event` was not passed
+            as the name.
+        """
+        if name is not MISSING and not isinstance(name, (str, Event)):
+            raise TypeError(
+                f"Bot.add_listener expected str or Enum but received {name.__class__.__name__!r} instead."
```

Comment:
since this is now also usable in `Client`s, this error message should only mention `add_listener` without the `Bot.` prefix - same thing in the other methods with similar error messages